### PR TITLE
feat(CSS): refactor CSS cluster resources to new sdk 

### LINKED ISSUE #1348: [CSS] add new resource: CSS cluster resources switch to sdk
1. add new resource: Word Dictionaries manage
2. maintain  snapshot resources

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
 refactor CSS cluster resources to new sdk 

**Which issue this PR fixes**:
fixes #1348 

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run=TestAccCssCluster_basic -timeout 360m -parallel 4
=== RUN   TestAccCssCluster_basic
=== PAUSE TestAccCssCluster_basic
=== CONT  TestAccCssCluster_basic
--- PASS: TestAccCssCluster_basic (1706.12s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css       1706.196s
```

### CODE REVIEW

File: docs/resources/css_cluster.md [hunk 1]
```diff
@@ -138,19 +147,33 @@ In addition to all arguments above, the following attributes are exported:
...
 
 This resource provides the following timeouts configuration options:
 
 * `create` - Default is 60 minute.
 * `update` - Default is 60 minute.
+* `delete` - Default is 60 minute.
+
+## Import
+
+CSS cluster can be imported by  `id`. For example,
```
Comment (Lance52259) [inline docs/resources/css_cluster.md]: there are two white space between `by` and `id`

File: huaweicloud/utils/tags.go [hunk 2]
```diff
@@ -38,6 +39,20 @@ func UpdateResourceTags(conn *golangsdk.ServiceClient, d *schema.ResourceData, r
 	return nil
 }
 
+//This is a help to query tags of resource, then set to state. The name must be: tags "css-cluster"
+func SetResourceTagsToState(d *schema.ResourceData, client *golangsdk.ServiceClient, resourceType string) error {
+	// set tags
+	if resourceTags, err := tags.Get(client, resourceType, d.Id()).Extract(); err == nil {
+		tagmap := TagsToMap(resourceTags.Tags)
+		if err := d.Set("tags", tagmap); err != nil {
```
Comment (chengxiangdong) [inline huaweicloud/utils/tags.go:47]: `err` has been declared above and can be used directly here.

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 3]
```diff
@@ -0,0 +1,680 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func ResourceCssCluster() *schema.Resource {
```
Comment (chengxiangdong) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go:21]: Suggest adding version of API to the function name.

Comment (chengxiangdong) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go:21]: Suggest adding version of API to the function name.

**Comment by chengxiangdong** [comment]
`huaweicloud/resource_huaweicloud_css_cluster_v1.go` and `huaweicloud/resource_huaweicloud_css_cluster_v1_test.go` will no longer be used. Do we delete them?

**Comment by chengxiangdong** [comment]
pls use `fmtp.DiagErrorf` insteadof `diag.Errorf`

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 4]
```diff
@@ -0,0 +1,680 @@
...
+		return diag.Errorf("Error creating CssClusterV1, err=%s", createErr)
+	}
+
+	clusterId := r.Cluster.Id
+
+	createResultErr := checkClusterCreateResult(ctx, cssV1Client, clusterId, d.Timeout(schema.TimeoutCreate))
+	if createResultErr != nil {
+		return diag.FromErr(createResultErr)
+	}
+	d.SetId(clusterId)
```
Comment (chengxiangdong) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go:254]: pls delete the commented code

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 5]
```diff
@@ -0,0 +1,680 @@
...
+
+	// enable snapshot function and set policy when "backup_strategy" was specified
+	// createBackupErr := resourceCssClusterCreateBackupStrategy(d, cssV1Client)
+	// if createBackupErr != nil {
+	// 	return diag.FromErr(createBackupErr)
+	// }
```
Comment (chengxiangdong) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: pls delete the commented code

File: huaweicloud/utils/tags.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/utils/tags.go:47]: this is a local scope var

**Comment by hmbig2** [comment]
> `huaweicloud/resource_huaweicloud_css_cluster_v1.go` and `huaweicloud/resource_huaweicloud_css_cluster_v1_test.go` will no longer be used. Do we delete them?

because snapshot resource use the old file,so delete them in other PR

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 6]
```diff
@@ -0,0 +1,680 @@
...
+			},
+			Volume: cluster.InstanceVolumeBody{
+				Size:       d.Get("node_config.0.volume.0.size").(int),
+				VolumeType: d.Get("node_config.0.volume.0.volume_type").(string),
+			},
+		},
+		Tags:                utils.ExpandResourceTags(d.Get("tags").(map[string]interface{})),
+		EnterpriseProjectId: config.GetEnterpriseProjectID(d),
+	}
+	//
```
Comment (chengxiangdong) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: suggest deleting `//`

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/css/resource_huaweicloud_css_cluster.go:21]: Suggest this resource  needs to shield the version of api. so keep it as it is

File: docs/resources/css_cluster.md [hunk 7]
```diff
@@ -24,17 +23,18 @@ resource "huaweicloud_css_cluster" "cluster" {
 
   node_config {
     flavor = "ess.spec-4u16g"
+    availability_zone = "{{ availability_zone }}"
```
Comment (Lance52259) [inline docs/resources/css_cluster.md]: use variables format

File: docs/resources/css_cluster.md [hunk 8]
```diff
@@ -127,6 +127,16 @@ The `backup_strategy` block supports:
...
 
+* `bucket` - (Optional, String) Specifies the OBS bucket used for index data backup. If there is snapshot data in an OBS
+   bucket, only the OBS bucket is used and cannot be changed.
+
+* `backup_path` - (Optional, String) Specifies the storage path of the snapshot in the OBS bucket.
+
+* `agency` - (Optional, String) Specifies the IAM agency used to access OBS.
+
+-> If the `bucket`, `backup_path`, and `agency` parameters are empty at the same time, the system will automatically
+   create an OBS bucket and IAM agent, otherwise the configured parameter values will be used.
```
Comment (Lance52259) [inline docs/resources/css_cluster.md]: Is `NOTE` ?

File: docs/resources/css_cluster.md [hunk 9]
```diff
@@ -138,19 +148,33 @@ In addition to all arguments above, the following attributes are exported:
...
 * `create` - Default is 60 minute.
 * `update` - Default is 60 minute.
+* `delete` - Default is 60 minute.
+
+## Import
+
+CSS cluster can be imported by `id`. For example,
+
+```
+terraform import huaweicloud_css_cluster.example abc123
```
Comment (Lance52259) [inline docs/resources/css_cluster.md]: please use a real ID

File: docs/resources/css_cluster.md (hunk 7 above)
Comment (hmbig2) [reply docs/resources/css_cluster.md]: ok

File: docs/resources/css_cluster.md [hunk 10]
```diff
@@ -138,19 +152,33 @@ In addition to all arguments above, the following attributes are exported:
 * `created` - Time when a cluster is created. The format is ISO8601:
   CCYY-MM-DDThh:mm:ss.
```
Comment (chengxiangdong) [inline docs/resources/css_cluster.md:153]: suggest to use `hh24`

Comment (hmbig2) [reply docs/resources/css_cluster.md:153]: This is a standard iso8601 format. so keep it as it is

Comment (chengxiangdong) [reply docs/resources/css_cluster.md:153]: ok，it should follow the ISO8601 format

**Pull request merged**